### PR TITLE
CI: CircleCI: Use --no-deps when installing packages in doc_requirements.txt

### LINKED ISSUE #20390: CircleCI test failures on main branch.
It fails because of a git version mismatch. CircleCI builds the correct Python3.8 wheel version, but that the installed version that Makefile sees is of the HEAD of the 1.21.x branch. Maybe the Makefile is using Python3.7? If so, I don't know why this is suddenly a problem.

### PR DESCRIPTION
**Pull request by WarrenWeckesser**
Closes gh-20390.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -45,7 +45,7 @@ jobs:
             pip install --progress-bar=off --upgrade pip 'setuptools<49.2.0'
             pip install --progress-bar=off -r test_requirements.txt
             pip install .
-            pip install --progress-bar=off -r doc_requirements.txt
+            pip install --progress-bar=off --no-deps -r doc_requirements.txt
```
Comment (mattip) [inline .circleci/config.yml:48]: Doesn't this mean no other packages like docutils will be installed with sphinx?

Comment (mattip) [reply .circleci/config.yml:48]: If so, we need two files, one with deps and one without

Comment (WarrenWeckesser) [reply .circleci/config.yml:48]: Right, I expect the first shot at this won't work.  Then the missing dependencies can be added to `doc_requirements.txt`, or to a new file that is to be used with `--no-deps` if we need to leave the current `doc_requirements.txt` untouched.

I don't claim this is a great solution.  If anyone has a better approach, I'm happy to close the PR.

**Comment by WarrenWeckesser** [comment]
> Maybe install NumPy last?

That could work, too.  But will `pip` complain that installing numpy version `1.23.0.dev0.[etc]` is not possible because scipy requires `<1.23.0`?  If so, is there a command-line option to turn off that check?

**Comment by WarrenWeckesser** [comment]
Closing to stop the tests.

File: .circleci/config.yml (hunk 1 above)
Comment (WarrenWeckesser) [reply .circleci/config.yml]: With sphinx, matplotlib, ipython and pandas (pandas is a doc requirement?), that adds up to a *lot* of dependent packages.  Let's try installing numpy last instead.

**Comment by WarrenWeckesser** [comment]
I forced-pushed a change with the PR closed, and now I can't reopen the PR.  New PR is https://github.com/numpy/numpy/pull/20393.

**Pull request closed without merging**